From david-b@pacbell.net Tue Aug  8 23:58:33 2006
From: Milan Svoboda <msvoboda@ra.rockwell.com>
Subject: USB gadget: gadgetfs dont try to lock before free
Date: Tue, 8 Aug 2006 22:14:43 -0700
To: Greg KH <greg@kroah.com>
Message-Id: <200608082214.43632.david-b@pacbell.net>

From: Milan Svoboda <msvoboda@ra.rockwell.com>

I spotted this during my tests with -rt on arm. The -rt patch contains
some better tools
to diagnose problems with locks and some other things...

Original code tries to take semaphore in BUG_ON and then free the memory
with this semaphore.


Signed-off-by: Milan Svoboda <msvoboda@ra.rockwell.com>
Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/gadget/inode.c |    1 -
 1 file changed, 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/gadget/inode.c
+++ gregkh-2.6/drivers/usb/gadget/inode.c
@@ -223,7 +223,6 @@ static void put_ep (struct ep_data *data
 	/* needs no more cleanup */
 	BUG_ON (!list_empty (&data->epfiles));
 	BUG_ON (waitqueue_active (&data->wait));
-	BUG_ON (down_trylock (&data->lock) != 0);
 	kfree (data);
 }
 
